c_src/ep_decoder.c

```diff
@@ -100,6 +100,44 @@ unpack_int32(ErlNifEnv *env, ep_dec_t *dec, ERL_NIF_TERM *term)
     return_error(env, dec->term);
 }
 
+static inline char * toArray(long int number, size_t size)
+{
+    int32_t i;
+    char *numberArray = calloc(size, sizeof(char));
+    for (i = size-1; i >= 0; --i, number /= 10)
+    {
+        numberArray[i] = (number % 10) + '0';
+    }
+    return numberArray;
+}
+
+static inline ERL_NIF_TERM
+unpack_int64_as_bin(ErlNifEnv *env, ep_dec_t *dec, ERL_NIF_TERM *term)
+{
+    int32_t     shift = 0, left = 10;
+    int64_t     val = 0;
+    ErlNifBinary    bin;
+
+    while (left && dec->p < dec->end) {
+
+        val |= ((uint64_t) (*(dec->p) & 0x7f) << shift);
+        if ((*(dec->p)++ & 0x80) == 0) {
+            size_t size = log10(val) + 1;
```

Comment:
calling log10 at each iteration seems expensive to me. I think give we are doing this for int64 we should basically allocate 20 bytes

Reply:
no, we are not calling log10 at each iteration.
we only call it at the end when we have the actual integer.

Reply:
Even one call to log10 is not worth trying to allocate the right amount of mem, Just allocating 32 bytes is probably best. @jg513 proposal where this memory is on the stack is even better.

Reply:
@NikolaBorisov got it, sure. updated it.
i asked the author of that repo to review my code - just to make sure we are all good.
please take a look again.